securedrop/server.py

```diff
@@ -230,6 +250,29 @@ async def add_contact(self, addc, stream):
         msg = self.users.add_contact(self.sock_to_email[stream], addc.name, addc.email)
         await self.write_status(stream, msg)
 
+    async def list_contacts(self, stream):
+        # three verification steps
+        current_user_email = self.sock_to_email[stream]
+        # 1: contacts_dict contains the names and email adresses that a user has added
+        contacts_dict = self.users.list_contacts(current_user_email)
+        contacts_dict_list = list(contacts_dict.keys())
+        online_users_list = list(self.email_to_sock.keys())
```

Comment:
This is unnecessary.

Reply:
cleaned up

Reply:
This isn't resolved. You don't need this variable at all, and converting it to a list just slows down everything since it has to linear search. Below you can just check if the email is in it with `if email in self.email_to_sock`.

Reply:
changed